sanic/response.py

```diff
@@ -59,16 +59,23 @@ class StreamingHTTPResponse(BaseHTTPResponse):
         "status",
         "content_type",
         "headers",
+        "chunked",
         "_cookies",
     )
 
     def __init__(
-        self, streaming_fn, status=200, headers=None, content_type="text/plain"
+        self,
+        streaming_fn,
+        status=200,
+        headers=None,
+        content_type="text/plain",
+        chunked=None,
```

Comment:
Wouldn't it make more sense in the way you're using this variable to just make it a boolean instead of originally setting it to `None`?

Reply:
I agree.

And, also instead of this:

```chunked = self.chunked
        self.headers.pop("Content-Length", None)
        if chunked is None:
            chunked = version != "1.0"

        if chunked:
```

Why not something simpler without any variable assignment?

```
if self.chunked and version != "1.0"
```

Reply:
I agree.

And, also instead of this:

```
        chunked = self.chunked
        self.headers.pop("Content-Length", None)
        if chunked is None:
            chunked = version != "1.0"

        if chunked:
```

Why not something simpler without any variable assignment?

```
if self.chunked and version != "1.0"
```

Reply:
@ahopkins `version != "1.0"` not possible because `version` is not available in the `write` method

Reply:
@ahopkins in general, I think there is a fundamental problem with `HTTPResponse` instances: they don't know an exact HTTP version (and some other info about the request) until the response is sent to the client. It forces me to use weird and complicated code like this assignment inside the `stream` method instead of `__init__`

Reply:
@seemethere @ahopkins ok I tried to simplify it https://github.com/huge-success/sanic/pull/1560/commits/4a7d30ee6c2937e5e6131616306448042254192e

Reply:
@seemethere @ahopkins ok I tried to simplify it https://github.com/huge-success/sanic/pull/1560/commits/7d6e60ab7dbe2ab92a451cf75bb8ca0a4eb69efb